steps/test/unit/tReorderMSData.cc

```diff
@@ -1,94 +1,108 @@
 
-#include <fstream>
 #include <algorithm>
+#include <fstream>
+#include <limits>
+
 #include <boost/test/unit_test.hpp>
 
 #include "tStepCommon.h"
-#include <dp3/steps/Step.h>
-#include "../../NullStep.h"
+#include "dp3/steps/Step.h"
 #include "../../InputStep.h"
+#include "../../NullStep.h"
 #include "../../ReorderMSData.h"
-#include <dp3/base/DPBuffer.h>
-#include <dp3/base/DPInfo.h>
 #include "../../../common/ParameterSet.h"
-#include "../../../common/StringTools.h"
 
-using dp3::base::DPBuffer;
-using dp3::base::DPInfo;
 using dp3::common::Fields;
 using dp3::common::ParameterSet;
 using dp3::steps::Step;
 
 BOOST_AUTO_TEST_SUITE(reorder)
 
-bool compareBinaries(const std::string& filename1,
-                     const std::string& filename2) {
-  std::ifstream file1(
-      filename1,
-      std::ifstream::ate | std::ifstream::binary);  // open file at the end
-  std::ifstream file2(
-      filename2,
-      std::ifstream::ate | std::ifstream::binary);  // open file at the end
-
-  if (file1.fail() || file2.fail()) {
-    std::cout << "Failed to open file" << std::endl;
-  }
+// Function to compare two binary files. If the files for comparison are Meta
+// file, the flag is_meta_file is set to true.
+void compareBinaryFiles(const std::string& reference_filename,
+                        const std::string& input_filename, bool is_meta_file) {
+  
+  std::ifstream reference_file(reference_filename);
+  std::ifstream input_file(input_filename);
 
-  const std::ifstream::pos_type fileSize = file1.tellg();
-  if (fileSize != file2.tellg()) {
-    return false;  // different file size
+  if (reference_file.fail() || input_file.fail()) {
+    BOOST_CHECK(false);
   }
 
-  file1.seekg(0);  // rewind
-  file2.seekg(0);  // rewind
+  // Meta files contain path information in the first line of the binary.
+  // Since the path representation differs (i.e absolute path vs relative path),
+  // we skip the path check and compare only the rest of the meta information.
+  if (is_meta_file) {
+    reference_file.ignore(std::numeric_limits<std::streamsize>::max(),
+                              '\n');
+    input_file.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+    if (reference_file.eof() || input_file.eof()) {
+      BOOST_CHECK(false);
+    }
```

Comment:
This would be better as:

```BOOST_CHECK(!reference_file.eof())
BOOST_CHECK(!input_file.eof())```

or similar

Reply:
Oh yes, that's silly that I wrote a if-statement for this. Thanks for pointing out. Modified all boost checks.